### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by mlilien**
elbb/bb-buildingblock#8 document prerequisites
elbb/bb-buildingblock#17 document how to use bb-buildingblock template
elbb/bb-buildingblock#3 elbb/bb-buildingblock#15 document dependancy/usage bb-gitversion

**blocked by:** elbb/bb-gitversion#26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,31 @@
...
+```
+
+Adapt your own ci/cd pipeline. Use the pipeline from `example/ci` directory and adapt it. E.g.:
+
+```bash
+cp -f example/ci/pipeline.yaml pipeline.yaml
+cp -rf example/ci/ci/* ci/
+```
+Follow "[Using concourse CI for a CI/CD build](#using-concourse-ci-for-a-cicd-build)" to adapt your pipeline configuartion.
+
```
Comment (siredmar) [inline README.md]: I guess many parts you would write are already present in the ci/cd integration guide readme.

File: README.md [hunk 2]
```diff
@@ -39,6 +60,14 @@ After successfully uploading the pipeline to concourse CI login and unpause it.
 
 See the [integration documentation](README_CICD_INTEGRATION.md) on how to modify the `pipeline.yaml` and config files for your building block.
 
+## Versioning
+
+This template building block uses [bb-gitversion](https://github.com/elbb/bb-gitversion) to compute the semantic version of itself from the git history.
+
+Using `dobi.sh` the version is updated when necessary and published in environment variables. You can use these environment variables in your dobi targets. 
+See `./gen/gitversion/env/gitversion.env` and `./gen/gitversion/branch/env/gitversion.env` for a detailed list of version environment variables. The later file `./gen/gitversion/branch/env/gitversion.env` includes variables which are formated for `master` and `non-master` branches differently. Currently it includes the variable `GitVersionBranch`. It is identically to `GitVersion_FullSemVer` in case of using the master branch. On all other branches it is identically to `GitVersion_InformationalVersion`.
```
Comment (siredmar) [inline README.md]: typo: later -> latter

File: README.md [hunk 3]
```diff
@@ -39,6 +60,14 @@ After successfully uploading the pipeline to concourse CI login and unpause it.
 
 See the [integration documentation](README_CICD_INTEGRATION.md) on how to modify the `pipeline.yaml` and config files for your building block.
 
+## Versioning
+
+This template building block uses [bb-gitversion](https://github.com/elbb/bb-gitversion) to compute the semantic version of itself from the git history.
+
+Using `dobi.sh` the version is updated when necessary and published in environment variables. You can use these environment variables in your dobi targets.
```
Comment (siredmar) [inline README.md:106]: ... and published in different formats e.g. environment variables or plain files containing version version information.

Comment (mlilien) [reply README.md:106]: how can you use these plain files in dobi? you mean via mount?

Comment (siredmar) [reply README.md:106]: you'll need them at least for concourse

File: README.md (hunk 1 above)
Comment (mlilien) [reply README.md]: removed the copy example

File: README.md [hunk 4]
```diff
@@ -4,6 +4,27 @@
...
+
+## Howto use bb-buildingblock template
+
+Create a new repository and merge bb-buildingblock, e.g.:
+```bash
+git init myrepo
+cd myrepo
+git remote add bb-buildingblock https://github.com/elbb/bb-buildingblock.git
+git fetch -n bb-buildingblock
+git merge bb-buildingblock master
```
Comment (siredmar) [inline README.md]: typo: `git merge bb-buildingblock/master`

**Review by siredmar — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mlilien) [reply README.md]: seems i have to rewrite it anyway

File: README.md [hunk 5]
```diff
@@ -39,6 +99,27 @@ After successfully uploading the pipeline to concourse CI login and unpause it.
...
 
+## Versioning
+
+This template building block uses [bb-gitversion](https://github.com/elbb/bb-gitversion) to compute the semantic version of itself from the git history on the bases of git tags.
+
+Using `dobi.sh` the version is updated when necessary and published in environment variables. You can use these environment variables in your dobi targets. 
+See `./gen/gitversion/env/gitversion.env` for a detailed list of version environment variables. 
+These key-value pairs are also generated into other representations:
+```
+./gen/gitversion/cpp/ellb/version.h    #c++ header
```
Comment (JoergZeidler) [inline README.md]: typo ellb -> elbb

**Review by JoergZeidler — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (mlilien) [reply README.md]: fixed

**Comment by mlilien** [comment]
@siredmar   Can you resolve your remarks above?

**Review by JoergZeidler — DISMISSED** [review]

**Review by siredmar — DISMISSED** [review]

**Pull request merged**